### PR TITLE
Create 2021-04-14-removing-implicit-coercions.markdown

### PR DESCRIPTION
**Pull request by DavidSnider**
Add blog post for implicit coercions

### CODE REVIEW

File: _posts/2021-04-14-removing-implicit-coercions.markdown [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+### Bitwise operations (`&`, `|`, `^`, `~`, `<<`, `>>`) and assignment equivalents
+
+**Current Behaviour:**
+
+`&`, `|`, `^`: null, bool, float, vec, dict, keyset, and resource will implicitly coerce to int. strings will implicitly coerce as well *unless* both operands are strings.
+
+```
+null & vec[]; // 0
+keyset[1] & true; // 1
+'abcd' & 'efgh'; // 'abc`'
```
Comment (fredemmott) [inline _posts/2021-04-14-removing-implicit-coercions.markdown:64]: might be worth clarifying this one: ascii('d') & ascii('h') == ascii('`'), or link to https://3v4l.org/CTH2o

File: _posts/2021-04-14-removing-implicit-coercions.markdown [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+```
+null == false; // true
+false == new Foo(); // false
+false == Foo::class; // false
+true == Foo::class; // true
+```
+
+**Future Behaviour:**
+There will be two steps to this, going through both Hard Runtime Enforcement and then New Behaviour. comparing values of two different types will throw an exception. Then, comparing values of two different types will simply be false. At the
```
Comment (fbnav) [inline _posts/2021-04-14-removing-implicit-coercions.markdown:191]: The new sentence starts with "comparing" and might have had an abrupt end after "At the"

Comment (DavidSnider) [reply _posts/2021-04-14-removing-implicit-coercions.markdown:191]: yeah this is a really weird error

**Comment by fredemmott** [comment]
squashed and merged as e11bd1d29d5accd90087d462f8a2898f032ff2f3

**Pull request closed without merging**